websites/O/OtakuEyes/presence.ts

```diff
@@ -0,0 +1,43 @@
+const presence = new Presence({
+		clientId: "784067980086607952",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+function checkFooterContent() {
+	const pElement = document.querySelector("footer p");
+	return pElement && pElement.textContent.trim() === "OtakuEyes";
+}
+
+presence.on("UpdateData", async () => {
+	if (checkFooterContent()) {
+		const [page, , episode] = document.location.pathname.split("/").slice(1, 4),
+			showCover = await presence.getSetting<boolean>("cover"),
+			presenceData: PresenceData = {
+				largeImageKey: "https://i.imgur.com/b3H1vRU.png",
+				startTimestamp: browsingTimestamp,
+			};
+
+		switch (page) {
+			case "anime": {
+				presenceData.details = "Watching an anime";
+				presenceData.state = episode
+					? `${document
+							.querySelector("h3.titleA")
+							.textContent.trim()}  - Episode ${episode}`
+					: document.querySelector("h3.titleA").textContent.trim();
+				if (showCover) {
+					presenceData.largeImageKey = document
+						.querySelector(".cover")
+						.getAttribute("src");
+				}
+				break;
+			}
+			default:
+				presenceData.details = "Looking for an anime to watch";
+				presenceData.state = "Home";
+				presenceData.smallImageKey = Assets.Search;
+		}
+
+		presence.setActivity(presenceData, true);
+	}
+});
```

Comment:
```suggestion
const presence = new Presence({
		clientId: "784067980086607952",
	}),
	browsingTimestamp = Math.floor(Date.now() / 1000);

function checkFooterContent() {
	const pElement = document.querySelector("footer p");
	return pElement && pElement.textContent.trim() === "OtakuEyes";
}

if (!checkFooterContent()) return;

presence.on("UpdateData", async () => {
		const [page, , episode] = document.location.pathname.split("/").slice(1, 4),
			showCover = await presence.getSetting<boolean>("cover"),
			presenceData: PresenceData = {
				largeImageKey: "https://i.imgur.com/b3H1vRU.png",
				startTimestamp: browsingTimestamp,
			};

		switch (page) {
			case "anime": {
				presenceData.details = "Watching an anime";
				presenceData.state = episode
					? `${document
							.querySelector("h3.titleA")
							.textContent.trim()}  - Episode ${episode}`
					: document.querySelector("h3.titleA").textContent.trim();
				if (showCover) {
					presenceData.largeImageKey = document
						.querySelector(".cover")
						.getAttribute("src");
				}
				break;
			}
			default:
				presenceData.details = "Looking for an anime to watch";
				presenceData.state = "Home";
				presenceData.smallImageKey = Assets.Search;
		}

		presence.setActivity(presenceData, true);
});
```

Reply:
Error: A 'return' statement can only be used within a function body.
Error: Error: presence.ts:11:28 - Error TS1108: A 'return' statement can only be used within a function body.
Error: Some Presences failed to compile, exiting...
Error: Process completed with exit code 1.

Reply:
A possible solution could be to wrap the listener inside the if statement.

Reply:
I will keep my code or maybe you have something better ?

Reply:
Rather than checking the content within the `presence.on("UpdateData", ...)` loop, put the `presence.on("UpdateData", ...)` inside the check, thus the check runs once and no wasted resources are used.
```ts
if (checkFooterContent()) {
  presence.on("UpdateData", () => {
    // ...
  });
}
```

Reply:
Oh yea sorry my fault, @theusaf is right

Reply:
Change done 👍 
Thank for the help